RFunction.R

```diff
@@ -7,279 +7,233 @@ library(magic)
 library(geosphere)
 library(lubridate)
 
-rFunction <-function(data, threshold=NULL, window=72, yaxs_limit=1000){
-  
-   data <- data |> mutate(location_long = sf::st_coordinates(data)[,1],
-                         location_lat = sf::st_coordinates(data)[,2],
-                         trackID = mt_track_id(data),
-                         distance = mt_distance(data))
+# TODO: 
+# - deal with test data having periods in column headers
 
-   #data$distance <- mt_distance(data, units="m") # like this units will always be in meters, please adjust to the units you see most fit
-  class(data$distance) <-"numeric"
-  data_df <-as.data.frame(data)
-  names(data_df) <- gsub("[.]", "_", names(data_df))
-  # uid <-unique(data_df$individual_local_identifier)
-  uid <-unique(data$trackID) 
-  
-  ###  Function for plotting the individual speed
-  plot_speed <-function(dat, dat_outp, yul=yaxs_limit)
+###  Function for plotting the individual speed
+plot_speed <- function(dat, dat_outp, yul, track_id, threshold) {
```

Comment:
Move plotting functions outside of main function to make main function more readable